pkg/client/tx/client.go

```diff
@@ -89,7 +89,11 @@ type (
 // the transactions subscription.
 type TxEvent struct {
 	// Tx is the binary representation of the tx hash.
-	Tx     []byte            `json:"tx"`
+	Tx     []byte   `json:"tx"`
+	Result TxResult `json:"result"`
+}
+
+type TxResult struct {
```

Comment:
I'm worried that `TxResult` may be conflated with the definition of a TxResult in Comet/Tendermint land. Wdyt if `ABCIEvents`, `TxResultEvents` or removing the decoupling altogether?

<img width="623" alt="Screenshot 2023-12-04 at 3 07 34 PM" src="https://github.com/pokt-network/poktroll/assets/1892194/6bc12534-89db-4a2a-b172-c3266e5c2ffc">

https://github.com/cosmos/cosmos-sdk/blob/main/proto/tendermint/abci/types.proto#L416

Reply:
Thanks for calling this out! :raised_hands:  This is indeed the correct type and we should be using it instead of the one(s) currently defined in our tx client package. :mechanical_arm: 

I have gone digging (on more than one occasion) for the actual protobuf that is being used by comet/cosmos to produce the JSON structure we see in the event subscription stream. I can't remember if I looked at this one before or not but I'm realizing that it's tricky to connect the dots because it's being serialized to JSON. :thinking:  Anyways, you nailed it - thanks!